File: GardenTracker.Core/Entities/Expense.cs
using GardenTracker.Core.Enums;

namespace GardenTracker.Core.Entities;

public class Expense
{
    public int Id { get; set; }
    public int SeasonId { get; set; }
    public int? BedId { get; set; }
    public int? SupplierId { get; set; }
    public ExpenseCategory Category { get; set; }
    public string Description { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public DateOnly ExpenseDate { get; set; }

    public Season Season { get; set; } = null!;
    public RaisedBed? Bed { get; set; }
    public Supplier? Supplier { get; set; }
}

File: GardenTracker.Web/src/types/expense.ts
export type ExpenseCategory =
  | 'Seeds'
  | 'Transplants'
  | 'Soil'
  | 'Fertilizer'
  | 'PestControl'
  | 'BedMaterials'
  | 'Tools'
  | 'Maintenance'
  | 'Other'

export interface Expense {
  id: number
  seasonId: number
  bedId?: number | null
  bedName?: string | null
  supplierId?: number | null
  supplierName?: string | null
  category: ExpenseCategory
  description: string
  amount: number
  expenseDate: string
}

export interface CreateExpenseRequest {
  bedId?: number | null
  category: ExpenseCategory
  description: string
  amount: number
  expenseDate: string
}

export interface UpdateExpenseRequest {
  bedId?: number | null
  category: ExpenseCategory
  description: string
  amount: number
  expenseDate: string
}
